cmake/support_catalyst.cmake

```diff
@@ -21,8 +21,8 @@ macro(FindCatalyst target_name)
         )
         message(INFO " Building against local Catalyst - path: ${LIGHTNING_CATALYST_SRC_PATH} - GIT TAG: ${CATALYST_GIT_TAG}")
 
-        target_include_directories(${target_name} PUBLIC ${LIGHTNING_CATALYST_SRC_PATH}/runtime/lib/backend/common)
-        target_include_directories(${target_name} PUBLIC ${LIGHTNING_CATALYST_SRC_PATH}/runtime/include)
+        target_include_directories(${target_name} SYSTEM PUBLIC ${LIGHTNING_CATALYST_SRC_PATH}/runtime/lib/backend/common)
```

Comment:
For my understanding, could you please clarify why we need to add `SYSTEM` here?

Reply:
The code quality of external libraries should not have an impact on your build system. Tagging headers as system will instruct the compiler not to raise errors that arise purely from the included code.

Reply:
Cool! Thanks!